kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1061 @@
+      this > b -> this.shouldThrowMaximumExceptionOnBetween(a, b, tolerance)
+      else     -> throw IllegalStateException()
+    }
+  }
+  
+  private fun Double.shouldThrowMinimumExceptionOnBetween(a: Double, b: Double, tolerance: Double) {
+    val message = "$this should be bigger than $a"
+    shouldThrowExceptionOnBetween(a, b, tolerance, message)
+  }
+  
+  private fun Double.shouldThrowMaximumExceptionOnBetween(a: Double, b: Double, tolerance: Double) {
+    val message = "$this should be smaller than $b"
+    shouldThrowExceptionOnBetween(a, b, tolerance, message)
+  }
+  
+  
+  private fun Double.shouldThrowExceptionOnBetween(a: Double,
+                                                   b: Double,
+                                                   tolerance: Double,
+                                                   message: String = "$this should be smaller than $b and bigger than $a"
+  ) {
+    shouldThrowAssertionError(message,
+                              { this.shouldBeBetween(a, b, tolerance) },
+                              { this shouldBe between(a, b, tolerance) })
+  }
+  
+  private fun Double.shouldThrowExceptionOnNotBetween(a: Double,
+                                                      b: Double,
+                                                      tolerance: Double,
+                                                      message: String = "$this should not be smaller than $b and should not be bigger than $a"
+  ) {
+    
+    shouldThrowAssertionError(message,
+                              { this.shouldNotBeBetween(a, b, tolerance) },
+                              { this shouldNotBe between(a, b, tolerance) })
+  }
+  
+  private fun onDoubleRange(block: (double: Double, valueToVariate: Double) -> Unit) {
+    (0.1..1.0 step 0.2).forEach { valueToVariate ->
+      assertAll(300, nonMaxValueDoubles) {
+        block(it, valueToVariate)
+      }
+    }
+  }
+  
+  private infix fun ClosedRange<Double>.step(step: Double): Iterable<Double> {
+    val sequence = generateSequence(start) { previous ->
+      if (previous == Double.POSITIVE_INFINITY) return@generateSequence null
+      val next = previous + step
+      if (next > endInclusive) null else next
     }
+    return sequence.asIterable()
+  }
+  
+  private infix fun Double.shouldMatchLessThan(x: Double) {
+    this shouldBe lt(x)
+    this shouldBeLessThan x
+    this should beLessThan(x)
+    
+    this shouldThrowExceptionOnNotLessThan x
+  }
+  
+  private infix fun Double.shouldThrowExceptionOnNotLessThan(x: Double) {
+    shouldThrowAssertionError("$this should not be < $x",
+                              { this shouldNotBe lt(x) },
+                              { this shouldNotBeLessThan x },
+                              { this shouldNot beLessThan(x) })
+  }
+  
+  private infix fun Double.shouldNotMatchLessThan(x: Double) {
+    this shouldNotBe lt(x)
+    this shouldNotBeLessThan x
```

Comment:
If this fails, won't the next two lines not be executed?

Reply:
That's correct, but I don't think it's a bad thing. We want the test to fail when this fails

Reply:
We have `assertSoftly` for this case.